controllers/cloudstack_datacenter_controller_test.go

```diff
@@ -16,13 +16,15 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	"github.com/aws/eks-anywhere/controllers"
+	"github.com/aws/eks-anywhere/internal/test"
 	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
 	"github.com/aws/eks-anywhere/pkg/constants"
 	"github.com/aws/eks-anywhere/pkg/providers/cloudstack"
 	"github.com/aws/eks-anywhere/pkg/providers/cloudstack/decoder"
 )
 
 func TestCloudStackDatacenterReconcilerSetupWithManager(t *testing.T) {
+	test.MarkIntegration(t)
```

Comment:
this made me think
What if we made engineers **have** to call this/some other method in order to use envtest?

Right now this method just skips the test if the env var is not set. But if the engineer doesn't add this line, the test will compile and try to run.

We could however "require" this method. Right now all tests using envtest need a package level variable called `env`, that is setup from `TestMain`. What if we set a pattern where instead of accessing this variable directly, we call a method to retrieve it? That way engineers will "need" to call this method if they want access to envtest. Of course, they could still look for the package level variable, but if we name it something like `internalEnvtest` and add a decent comment, I doubt this will happen by mistake.

Quick example of what I'm proposing:

```go
func TestCloudStackDatacenterReconcilerSetupWithManager(t *testing.T) {
  env := requireEnvtest(t)
```

The in this package `main_test.go`:
```go
// internalEnvtest allows to use envtest when integration
// tests are enabled.
// DO NOT use it directly, instead use requireEnvtest.
var internalEnvtest *envtest.Environment

func requireEnvtest(tb testing.TB) *envtest.Environment {
  test.MarkIntegration(t)
  return internalEnvtest
}
```

Reply:
I'm not sold on the value. `envtest` features in a subset of integration tests. Moreover, developers have the freedom to tweak how envtest gets configured dependent on their usecase.